### ISSUE TITLE
The loss training log-likelihood when (x, y) is absolutely independent

### ISSUE THREAD

**Issue by longdvt** [Linear95/CLUB#26]
As I know, in section 3.2, we need to learn a variational distribution $q_\theta$(y|x), to approximate $p(y|x)$. If we could learn this distribution that is good enough, the MI estimator would good. How ever, when (x, y) is absolutely independent, we can not do it. It means the loss when training log-likelihood will explode. Did I understand it right ?

**Comment by Linear95** [comment]
This is an interesting perspective. I agree that if (x,y) is independent you may not learn q_theta(y|x) well. However, I don't believe the training loss will explode. This learning loss can be stable: if MI is not estimated well, the corresponding gradient will push (x,y) to update to have some unexpected correlation; then the situation changes to (x,y) are not independent, and the MI estimator works again. So I think the behaviors of the MI minimization near the independent point are within a stable equilibrium.

**Comment by longdvt** [comment]
In order to estimate MI effectively, Is it essential for learning log-likelihood loss need to reach to the value logvar (in case $q_\theta(y|x))$ is Gaussian)  or it just need to be stable ?

**Comment by Linear95** [comment]
It depends on what task you are working on. If you aim to estimate MI, a good variational approximation can make a better estimation. If you want to minimize the MI, the quality of your variational approximation is not that curial. We also tested the CLUBmean estimator, which can also work effectively with the assumption `logvar = 0`.

**Comment by longdvt** [comment]
I got it. Thanks for responding!
